### PR TITLE
New module: zabbix mediatype

### PR DESCRIPTION
**Pull request by rubentsirunyan**
##### SUMMARY
Created a new module for interacting with Zabbix mediatypes.
[API documentation](https://www.zabbix.com/documentation/4.0/manual/api/reference/mediatype/object)

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME

zabbix_mediatype

### CODE REVIEW

**Comment by rubentsirunyan** [comment]
This is not draft anymore, so feel free to review :)

File: lib/ansible/modules/monitoring/zabbix/zabbix_mediatype.py [hunk 1]
```diff
@@ -0,0 +1,688 @@
...
+    name: "My SMS Mediatype"
+    server_url: "http://example.com/zabbix/"
+    login_user: Admin
+    login_password: "zabbix"
+    type: 'sms'
+    gsm_modem: '/dev/ttyS0'
+'''
+
+try:
+    from zabbix_api import ZabbixAPI, ZabbixAPISubClass
```
Comment (D3DeFi) [inline lib/ansible/modules/monitoring/zabbix/zabbix_mediatype.py]:
```suggestion
    from zabbix_api import ZabbixAPI
```
Unused import

**Review by D3DeFi — APPROVED** [review]
This was such a joy to look at :) Awesome work @rubentsirunyan !

LGTM

**Comment by D3DeFi** [comment]
+1

File: lib/ansible/modules/monitoring/zabbix/zabbix_mediatype.py [hunk 2]
```diff
@@ -0,0 +1,688 @@
...
+        choices:
+            - USA
+            - Canada
+extends_documentation_fragment:
+    - zabbix
+
+'''
+
+RETURN = '''
+msg:
```
Comment (resmo) [inline lib/ansible/modules/monitoring/zabbix/zabbix_mediatype.py]: msg is a common return and must not be documented. please remove it like the following example

~~~
RETURN = r'''
'''
~~~

File: lib/ansible/modules/monitoring/zabbix/zabbix_mediatype.py [hunk 3]
```diff
@@ -0,0 +1,688 @@
...
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    if module.params['state'] == 'present':
+        validate_params(module, required_params)
+
+    if not HAS_ZABBIX_API:
+        module.fail_json(msg="Missing required zabbix-api module (check docs or install with: pip install zabbix-api)")
```
Comment (resmo) [inline lib/ansible/modules/monitoring/zabbix/zabbix_mediatype.py]: Please use the common helper for missing dependency errors, see https://github.com/ansible/ansible/blob/0f2903780bf8dfefb62247cad768f25858de7d73/lib/ansible/module_utils/cloudstack.py#L49 as an example

**Review by resmo — CHANGES_REQUESTED** [review]

**Review by resmo — APPROVED** [review]
shipit

**Comment by D3DeFi** [comment]
@resmo should I update other zabbix modules with above mentioned helper for missing deps?

**Comment by resmo** [comment]
@D3DeFi this should be used in all modules, yes

**Pull request merged**